pkg/tfgen/generate.go

```diff
@@ -1814,7 +1809,20 @@ func addRenamesToMetadataInfo(g *Generator) error {
 		return nil
 	}
 
-	renames, err := g.Renames()
+	var renames Renames
+
+	_, renamesAlreadySet, err := metadata.Get[Renames](g.info.MetadataInfo.Data, "renames")
+	if err != nil {
+		return fmt.Errorf("[pkg/tfgen] failed to retrieve renames from MetadataInfo: %w", err)
+	}
+
+	// Skip setting renames, for scenarios such as muxed providers that have them precomputed.
+	if renamesAlreadySet {
```

Comment:
Adding this early termination - this is important not to clobber in a multi-provider case. Sounds like if renames are already set adding them by tfgen should be either an error or it should resolve the conflict somehow (merge?). Here it resolves the conflict to the prior renames bit set.